clinical/src/main/kotlin/com/hartwig/actin/clinical/feed/standard/DataQualityMask.kt

```diff
@@ -44,9 +55,14 @@ fun ProvidedMolecularTestResult.isAllFieldsExceptGeneNull(): Boolean {
             this.vaf == null
 }
 
-class DataQualityMask(private val panelGeneList: PanelGeneList) {
+class DataQualityMask(private val panelGeneList: PanelGeneList, private val clinicalConfiguration: ClinicalConfiguration) {
     fun apply(ehrPatientRecord: ProvidedPatientRecord): ProvidedPatientRecord {
-        return ehrPatientRecord.scrubMedications().scrubModifications().addAlwaysTestedGenes(panelGeneList)
+        return ehrPatientRecord.scrubMedications()
+            .scrubModifications()
+            .addAlwaysTestedGenes(panelGeneList)
             .removeAllEmptyMolecularTestResults()
+            return if (clinicalConfiguration.useOnlyPriorOtherConditions) {
+                scrubbed.useOnlyPriorOtherConditions()
+            } else scrubbed
```

Comment:
I didn't mean for those suggestions to be a complete solution, but now I feel responsible 😛 
```suggestion
        return ehrPatientRecord.scrubMedications()
            .scrubModifications()
            .addAlwaysTestedGenes(panelGeneList)
            .removeAllEmptyMolecularTestResults().let { scrubbed ->
                if (clinicalConfiguration.useOnlyPriorOtherConditions) {
                    scrubbed.useOnlyPriorOtherConditions()
                } else scrubbed
            }
```

Reply:
Was just busy with something else so decided to take your changes and check them locally later :)